scripts/ccpp_track_variables.py

```diff
@@ -0,0 +1,194 @@
+from mkstatic import Suite
+from parse_checkers import registered_fortran_ddt_names
+from parse_tools import init_log, set_log_level
+from framework_env import CCPPFrameworkEnv
+
+###############################################################################
+# Set up the command line argument parser and other global variables          #
+###############################################################################
+
+###############################################################################
+# Functions and subroutines                                                   #
+###############################################################################
+
+def parse_arguments():
+    """Parse command line arguments."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-s', '--sdf', help='suite definition file to parse', required=True)
+    parser.add_argument('-m', '--metadata_path',
+                        help='path to CCPP scheme metadata files', required=True)
+    parser.add_argument('-c', '--config', 
+                        help='path to CCPP prebuild configuration file', required=True)
+    parser.add_argument('-v', '--variable', help='variable to track through CCPP suite', 
+                        required=True)
+    parser.add_argument('--debug', action='store_true', help='enable debugging output', 
+                        default=False)
+
+    args = parser.parse_args()
+
+    return(args)
+
+def setup_logging(debug):
+    """Sets up the logging module and logging level."""
+
+    #Use capgen logging tools
+    logger = init_log('ccpp_track_variables')
+
+    if debug:
+        set_log_level(logger, logging.DEBUG)
+        logger.info('Logging level set to DEBUG')
+    else:
+        set_log_level(logger, logging.WARNING)
+    return logger
+
+def parse_suite(sdf, run_env):
+    """Reads the provided sdf, parses into a Suite data structure, including the "call tree":
+       the ordered list of schemes for the suite specified by the provided sdf"""
+    run_env.logger.info(f'Reading sdf {sdf} and populating Suite object')
+    suite = Suite(sdf_name=sdf)
+    success = suite.parse()
+    if not success:
+        raise Exception(f'Parsing suite definition file {sdf} failed.')
+    run_env.logger.info(f'Successfully read sdf {suite.sdf_name}')
+    run_env.logger.info(f'Creating calling tree of schemes for suite {suite.name}')
+    success = suite.make_call_tree()
+    if not success:
+        raise Exception(f'Failed to create call tree of schemes for suite {suite.name}')
+    return suite
+
+def create_metadata_filename_dict(metapath):
+    """Given a path, read all .meta files in that directory and add them to a dictionary: the keys
+       are the name of the scheme, and the values are the filename of the .meta file associated
+       with that scheme"""
+
+    metadata_dict = {}
+    scheme_filenames=glob.glob(os.path.join(metapath, "*.meta"))
```

Comment:
Is there a reason why you don't get those from `ccpp_prebuild_config.py`?

Reply:
I don't really like using `glob`, and also the schemes may not all sit in the same directory. For example, the gsl folks have a chemistry fork where the chemistry metadata is in a separate subdirectory `chemistry` in the ccpp-physics repo.

Reply:
Regarding getting the metadata files from ccpp_prebuild_config.py, I was originally hoping I could eliminate the dependency on that file, and allow users to specify whichever metadata path(s) they would like.

Re: `glob`, is the issue of using `glob` specifically or that the current design does not allow users to specify multiple directories? If it is the former I will have to re-think how the metadata files are specified, but if it is the latter I can easily put in a loop to allow multiple directories to be specified on the command line. Let me know if you think the second approach is acceptable.

Reply:
Re. 1: that's ok, even if the path(s) contain schemes that are not in use by this model (i.e. not in the prebuild config), the suites won't have those schemes.

Re. 2. There is a general aversion to glob, more so for `cmake` than for other tools, but nonetheless. If it can be avoided, fine, but if you need it than that's ok for me. Yes, you will need a capability to have one or more metadata paths if you don't use the prebuild config. Note though that you put the burden on the user to know where all the files are located that may be used in the suite.